src/assets/wise5/components/outsideURL/outside-url-student/outside-url-student.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component } from '@angular/core';
+import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
+import { AnnotationService } from '../../../services/annotationService';
+import { ConfigService } from '../../../services/configService';
+import { NodeService } from '../../../services/nodeService';
+import { StudentDataService } from '../../../services/studentDataService';
+import { UtilService } from '../../../services/utilService';
+import { ComponentStudent } from '../../component-student.component';
+import { ComponentService } from '../../componentService';
+
+@Component({
+  selector: 'outside-url-student',
+  templateUrl: 'outside-url-student.component.html'
+})
+export class OutsideUrlStudent extends ComponentStudent {
+  url: SafeUrl;
+  infoUrl: SafeUrl;
+  infoString: string;
+  width: string;
+  height: string;
+
+  constructor(
+    protected AnnotationService: AnnotationService,
+    protected ComponentService: ComponentService,
+    protected ConfigService: ConfigService,
+    protected NodeService: NodeService,
+    private sanitizer: DomSanitizer,
+    protected StudentDataService: StudentDataService,
+    protected UtilService: UtilService
+  ) {
+    super(
+      AnnotationService,
+      ComponentService,
+      ConfigService,
+      NodeService,
+      StudentDataService,
+      UtilService
+    );
+  }
+
+  ngOnInit(): void {
+    this.url = this.getURL(this.componentContent);
+    this.infoUrl = this.getInfoUrl(this.componentContent);
+    this.infoString = this.getInfoString(this.componentContent);
+    this.width = this.getWidth(this.componentContent);
+    this.height = this.getHeight(this.componentContent);
+  }
+
+  getURL(componentContent: any): SafeUrl {
+    return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.url);
+  }
+
+  getInfoUrl(componentContent: any): SafeUrl {
+    if (componentContent.info == null || componentContent.info === '') {
+      return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.url);
+    } else {
+      return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.info);
+    }
```

Comment:
```suggestion
    if (this.isInfoSet(componentContent)) {
      return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.info);
    } else {
      return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.url);
    }
    
    function isInfoSet(comopnentContent: any): boolean {
      return componentContent.info != null && componentContent.info !== '';
    }
```

For this block and the getInfoString function, I think it'll become easier to read if we
1. define a isInfoSet function.
2. switch out the if and else condition statements, so ```return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.info);``` goes in the true condition case and ```return this.sanitizer.bypassSecurityTrustResourceUrl(componentContent.url);``` goes in the false condition case.

What do you think?